target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom

```diff
@@ -32,6 +32,7 @@ case "$FIRMWARE" in
 	dlink,dir-842-c2|\
 	dlink,dir-842-c3|\
 	dlink,dir-859-a1|\
+	dlink,dir-869-a1|\
 	nec,wg1200cr|\
 	wd,mynet-n750)
 		caldata_extract "art" 0x1000 0x440
```

Comment:
It's a good point to convert calibration data to nvmem, please refer 48625a04453b4f5775db2d2a5abebf977b81344c.
And now we can set ascii mac address in dts, please check if it works. https://github.com/openwrt/openwrt/commit/ecd81de7a5ab4899cc95107b402f7a402caf70bd https://github.com/openwrt/openwrt/commit/5c147d36ba7e998aee7ff4f81553b935bcfc5383 .

Reply:
It's worth to rebase upstream/master and try this patch:
<details>
<summary>Convert to nvmem</summary>

```
diff --git a/target/linux/ath79/dts/qca9563_dlink_dir-8x9-a1.dtsi b/target/linux/ath79/dts/qca9563_dlink_dir-8x9-a1.dtsi
index c5fcb0c8b0..36059d81b8 100644
--- a/target/linux/ath79/dts/qca9563_dlink_dir-8x9-a1.dtsi
+++ b/target/linux/ath79/dts/qca9563_dlink_dir-8x9-a1.dtsi
@@ -27,6 +27,14 @@
 
 &pcie {
 	status = "okay";
+
+	wifi@0,0 {
+		compatible = "qcom,ath10k";
+		reg = <0 0 0 0 0>;
+
+		nvmem-cells = <&calibration_ath10k>, <&macaddr_devdata_94>;
+		nvmem-cell-names = "pre-calibration", "mac-address-ascii";
+	};
 };
 
 &spi {
@@ -58,6 +66,18 @@
 				label = "devdata";
 				reg = <0x050000 0x10000>;
 				read-only;
+
+				compatible = "nvmem-cells";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				macaddr_devdata_94: macaddr@94 {
+					reg = <0x94 0x11>;
+				};
+
+				macaddr_devdata_b0: macaddr@b0 {
+					reg = <0xb0 0x11>;
+				};
 			};
 
 			partition@60000 {
@@ -72,10 +92,22 @@
 				reg = <0x070000 0xf80000>;
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
 				label = "art";
 				reg = <0xff0000 0x010000>;
 				read-only;
+
+				compatible = "nvmem-cells";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				calibration_ath9k: calibration@1000 {
+					reg = <0x1000 0x440>;
+				};
+
+				calibration_ath10k: calibration@5000 {
+					reg = <0x5000 0x844>;
+				};
 			};
 		};
 	};
@@ -115,5 +147,6 @@
 &wmac {
 	status = "okay";
 
-	qca,no-eeprom;
+	nvmem-cells = <&calibration_ath9k>, <&macaddr_devdata_b0>;
+	nvmem-cell-names = "calibration", "mac-address-ascii";
 };
diff --git a/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom b/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
index b5f3460652..0746b48290 100644
--- a/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
+++ b/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
@@ -31,8 +31,6 @@ case "$FIRMWARE" in
 	dlink,dir-842-c1|\
 	dlink,dir-842-c2|\
 	dlink,dir-842-c3|\
-	dlink,dir-859-a1|\
-	dlink,dir-869-a1|\
 	nec,wg1200cr|\
 	wd,mynet-n750)
 		caldata_extract "art" 0x1000 0x440
diff --git a/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index 3eb3272ab7..ffb62513e1 100644
--- a/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ b/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -61,11 +61,6 @@ case "$FIRMWARE" in
 		caldata_extract "art" 0x5000 0x844
 		ath10k_patch_mac $(mtd_get_mac_ascii bdcfg wlanmac_a)
 		;;
-	dlink,dir-859-a1|\
-	dlink,dir-869-a1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(mtd_get_mac_ascii devdata "wlan5mac")
-		;;
 	elecom,wrc-1750ghbk2-i)
 		caldata_extract "art" 0x5000 0x844
 		;;

```
</details>

Reply:
Thanks for the patch, I did the rebase and applied the patch you sent. Unfortunately, after flashing the image, I found out that only 5Ghz WiFi works. I am sending the log and I will try to find where the problem could be.
[kernel.log](https://github.com/openwrt/openwrt/files/9959585/kernel.log)

Reply:
Looks like you tested v22.03 branch? `mac-address-ascii` can only work in the current master branch. Maybe you can try to reset it and change the wifi country code.